dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientsPage.java

```diff
@@ -483,9 +487,15 @@ public boolean validateServerUrlOAI() {
 
             // First, we'll try to obtain the list of supported metadata formats:
             try {
-                List<String> formats = oaiHandler.runListMetadataFormats();
-                if (formats != null && formats.size() > 0) {
-                    createOaiMetadataFormatSelectItems(formats);
+                List<MetadataFormat> formats = oaiHandler.runListMetadataFormats();
+                if (formats != null && !formats.isEmpty()) {
```

Comment:
If we are okay with note above (that `oaiHandler.runListMetadataFormats();` shouldn't return `null` in any case) then we could omit `formats` null checking.

Reply:
Adapted it.